### PR TITLE
Mega-PR combining all the PRs

### PR DESCRIPTION
**Pull request by dan-simon**
This combines everything into one PR to avoid horrible merge conflicts and get community notations in the right folder (from now on I'm not going to do the work of putting community notations in the community folder for PR authors, though (unless of course it's my community notation)). It also adds another community notation. I'm mostly making it a PR right now to see the code quality problems.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -39,3 +39,15 @@ export function toEngineering(value: Decimal): Decimal {
 export function toFixedEngineering(value: Decimal, places: number): Decimal {
   return fixMantissaOverflow(toEngineering(value), places, 1000, 3);
 }
+
+const SUBSCRIPT_NUMBERS = ["₀", "₁", "₂", "₃", "₄", "₅", "₆", "₇", "₈", "₉"];
+
+export function toSubscript(num: number): string {
```
Comment (razenpok) [inline src/utils.ts]:
```suggestion
export function toSubscript(value: number): string {
```

File: src/utils.ts [hunk 2]
```diff
@@ -39,3 +39,15 @@ export function toEngineering(value: Decimal): Decimal {
...
+  return num.toFixed(0).split("").map(x => SUBSCRIPT_NUMBERS[parseInt(x)]).join("");
+}
+
+const SUPERSCRIPT_NUMBERS = ["⁰", "¹", "²", "³", "⁴", "⁵", "⁶", "⁷", "⁸", "⁹"];
+
+export function toSuperscript(num: number): string {
```
Comment (razenpok) [inline src/utils.ts]:
```suggestion
export function toSuperscript(value: number): string {
```

File: src/notation.ts [hunk 3]
```diff
@@ -12,26 +12,44 @@ export abstract class Notation {
...
 
+  public get negativeInfinite(): string {
+    return `-${this.infinite}`;
+  }
+
   public get infinite(): string {
     return "Infinite";
   }
 
+  public formatNegativeVerySmallDecimal(value: Decimal, places: number): string {
```
Comment (razenpok) [inline src/notation.ts]:
```suggestion
  public formatVerySmallNegativeDecimal(value: Decimal, places: number): string {
```

File: src/notation.ts [hunk 4]
```diff
@@ -12,26 +12,44 @@ export abstract class Notation {
...
+     return `-${this.formatVerySmallDecimal(value, places)}`;
+  }
+
+  public formatVerySmallDecimal(_value: Decimal, places: number): string {
```
Comment (razenpok) [inline src/notation.ts]:
```suggestion
  public formatVerySmallDecimal(value: Decimal, places: number): string {
```

File: src/community/precise-prime.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+export class PrecisePrimeNotation extends Notation {
+  public get name(): string {
+    return "Precise Prime";
+  }
+
+  public get infinite(): string {
+    return "Primefinity?";
+  }
+
```
Comment (razenpok) [inline src/community/precise-prime.ts:20]: Excess empty line

**Review by razenpok — CHANGES_REQUESTED** [review]
I didn't nitpick performance or structure of community notations since they are not a part of AD. I'm leaving them to your discretion.

I'm also awaiting a response from AD team regarding #41, so I'd suggest waiting for a bit before merging

File: src/notation.ts (hunk 4 above)
Comment (dan-simon) [reply src/notation.ts]: I'm not sure if changing 0 to value.toNumber() was part of this, but it seemed wise.

**Comment by dan-simon** [comment]
I'd appreciate nitpicks on community notations if you want to provide them and have the time. I'd especially like to know if Precise Prime is currently too slow to be in the main game (of course I'm not going to replace Prime with it without strong tester consensus including you and Iker, but in case it does eventually replace Prime it seems important to know).

**Comment by razenpok** [comment]
Sorry, but I think in case of me actually having some spare time, I'd better put it into AD development (still in hopes to release it somewhere during 2020).

Of course if we suddenly decide to incorporate some notation into base pack, it will undergo an additional performance and code style review. Until that it's fine if the community notations are not up to the _highest_ quality we can achieve (just keep them clean and reasonable).

**Review by razenpok — APPROVED** [review]

**Pull request merged**